### PR TITLE
Add USE_IPV4 option to make it easy to turn on IPv6 support

### PR DESCRIPTION
**Pull request by stbenjam**
This makes it easy to turn on IPv6.  Just set `USE_IPV4=false`.

### CODE REVIEW

**Comment by stbenjam** [comment]
Let's see how badly CI complains

**Comment by russellb** [comment]
I expect this will break because we still have to edit the RHCOS image used for the bootstrap VM.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1557/

**Comment by stbenjam** [comment]
> I expect this will break because we still have to edit the RHCOS image used for the bootstrap VM.

I saw the note in the doc but wasn't sure if it was fixed in 4.4+. It's failing earlier than that anyway:


```
Mar 02 20:01:44 |  [0;31mThe full traceback is:[0m
Mar 02 20:01:44 |  [0;31mWARNING: The below traceback may *not* be related to the actual failure.[0m
Mar 02 20:01:44 |  [0;31m  File "/tmp/ansible_virt_net_payload_i2pV3t/ansible_virt_net_payload.zip/ansible/modules/cloud/misc/virt_net.py", line 628, in main[0m
Mar 02 20:01:44 |  [0;31m  File "/tmp/ansible_virt_net_payload_i2pV3t/ansible_virt_net_payload.zip/ansible/modules/cloud/misc/virt_net.py", line 522, in core[0m
Mar 02 20:01:44 |  [0;31m  File "/tmp/ansible_virt_net_payload_i2pV3t/ansible_virt_net_payload.zip/ansible/modules/cloud/misc/virt_net.py", line 429, in start[0m
Mar 02 20:01:44 |  [0;31m  File "/tmp/ansible_virt_net_payload_i2pV3t/ansible_virt_net_payload.zip/ansible/modules/cloud/misc/virt_net.py", line 419, in create[0m
Mar 02 20:01:44 |  [0;31m  File "/tmp/ansible_virt_net_payload_i2pV3t/ansible_virt_net_payload.zip/ansible/modules/cloud/misc/virt_net.py", line 217, in create[0m
Mar 02 20:01:44 |  [0;31m  File "/usr/lib64/python2.7/site-packages/libvirt.py", line 2990, in create[0m
Mar 02 20:01:44 |  [0;31m    if ret == -1: raise libvirtError ('virNetworkCreate() failed', net=self)[0m
Mar 02 20:01:44 |  [0;31m[0m
Mar 02 20:01:44 |  [0;31mfailed: [localhost] (item={u'bridge': u'baremetal', u'domain': u'ostest.test.metalkube.org', u'forward_mode': u'nat', u'name': u'baremetal', u'netmask': u'ffff:ffff:ffff:ffff:ffff:ffff:ffff:ff00', u'prefix': u'120', u'dhcp_range': [u'fd2e:6f44:5dd8:c956::14', u'fd2e:6f44:5dd8:c956::3c'], u'dns': {u'hosts': [{u'ip': u'fd2e:6f44:5dd8:c956::5', u'hostnames': [u'api']}, {u'ip': u'fd2e:6f44:5dd8:c956::2', u'hostnames': [u'ns1']}, {u'ip': u'fd2e:6f44:5dd8:c956::1', u'hostnames': [u'virthost']}], u'forwarders': [{u'domain': u'apps.ostest.test.metalkube.org', u'addr': u'::1'}]}, u'address': u'fd2e:6f44:5dd8:c956::1', u'nat_port_range': [1024, 65535]}) => {[0m
```

**Comment by russellb** [comment]
I tested both 4.4 and 4.5/master yesterday, and they were both broken for different reasons.  I got further than the error in CI, though.

**Comment by stbenjam** [comment]
You're testing on EL8? We might have to wait for CI to be EL8 to make it the default.

**Comment by hardys** [comment]
> I saw the note in the doc but wasn't sure if it was fixed in 4.4+. It's failing earlier than that anyway:

It's not yet fixed - the latest attempt to resolve it is tracked via https://bugzilla.redhat.com/show_bug.cgi?id=1803926 then we'll need that dracut in a new RHCOS image and bumps for 4.5/4.4/4.3 installer.

**Comment by russellb** [comment]
> You're testing on EL8? We might have to wait for CI to be EL8 to make it the default.

yes, i'm using rhel 8

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1591/

**Comment by russellb** [comment]
> Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1591/

```
    "msg": "internal error: Check the host setup: enabling IPv6 forwarding with RA routes without accept_ra set to 2 is likely to cause routes loss. Interfaces to look at: eno2"
```

This is a host configuration issue on the CI host.  We should probably script it anyway.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1592/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1593/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1594/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1595/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1596/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1597/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1598/

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1599/

File: rhcos.sh [hunk 1]
```diff
@@ -23,3 +23,54 @@ export MACHINE_OS_BOOTSTRAP_IMAGE_SHA256=${MACHINE_OS_BOOTSTRAP_IMAGE_SHA256:-${
...
+    set -e
+    if [[ -n "${IP_PARAM}" ]] ; then
+        # Image already successfully hacked
+        gzip "${UNCOMPRESSED_IMAGE_FILE}"
+        popd
+        return
+    fi
+
+    # Fix the ip= kernel command line to work for IPv6.  This is temporary.
+    virt-edit -a "${UNCOMPRESSED_IMAGE_FILE}" -m /dev/sda1 -e "s/ip=dhcp/ip=ens3:dhcp6/g" /grub2/grub.cfg
```
Comment (hardys) [inline rhcos.sh]: https://github.com/openshift/installer/commit/9ed508a3a86da759f68c1c55d43b48593c3e2ca4 landed with the fix for https://bugzilla.redhat.com/show_bug.cgi?id=1803926 so we may be able to remove this now (and I'm not sure the virt-edit will work with the new defaults)

Comment (derekhiggins) [reply rhcos.sh]: If the virt-edit needs to stay we may need to figure out a way to edit the image and take a copy without changing the original image(or do caching differently), this image edit is changing the file in place which is one of the few directories that are persisted between jobs.

Comment (hardys) [reply rhcos.sh]: I tested locally and I think the image hack can now be removed for 4.5 builds - I guess we may need to manually purge any modified images though?

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1603/

File: rhcos.sh (hunk 1 above)
Comment (russellb) [reply rhcos.sh]: I disabled the hack for 4.5.

Good point about editing in-place and that breaking the cache in CI!  Now that this doesn't run for 4.5 though, this CI system isn't going to run it anymore ...

Comment (derekhiggins) [reply rhcos.sh]: Ack, I'll purge the image cache in CI and we should be ok for future patches.

**Comment by metal3ci** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/1607/

**Pull request merged**